data-prepper-plugins/otel-proto-common/src/test/java/org/opensearch/dataprepper/plugins/otel/codec/OTelProtoStandardCodecTest.java

```diff
@@ -218,8 +219,17 @@ private void validateSpans(final List<Span> spans) {
             assertThat(spans.size(), is(equalTo(3)));
 
             for (final Span span : spans) {
-                assertThat(span.getTraceGroup(), nullValue());
-                assertThat(span.getTraceGroupFields(), nullValue());
+                if (span.getParentSpanId().isEmpty()) {
```

Comment:
This is a somewhat loose assertion. Do we even know if we ever get non-empty spans?

Reply:
This is the exact same code that exists in OTelProtoOpensearchCodecTest. I did not include previously because there was not tracegroups support previously.

Reply:
Yea, I see that you have expanded the scope of this test.

Could it be split into two different tests? One for each scenario?

Reply:
@dlvenable  The test data is coming from `resources/test-request.json` and it has two spans with parentSpan and one without parentSpan. So, I think this test is fine.